File: querydsl-scala/src/test/scala/com/querydsl/scala/GenericExporterTest.scala
package com.querydsl.scala

import java.io.File

import com.querydsl.codegen.GenericExporter
import org.junit.Test

class GenericExporterTest {

  @Test
  def Export {
    val exporter = new GenericExporter()
    exporter.setTargetFolder(new java.io.File("target/gen1"))
    exporter.setSerializerClass(classOf[ScalaEntitySerializer])
    exporter.setTypeMappingsClass(classOf[ScalaTypeMappings])
    exporter.setCreateScalaSources(true)
    exporter.export(getClass.getPackage)

    val targetFolder = new File("target/gen1")
    CompileTestUtils.assertCompileSuccess(targetFolder)
  }

}

File: querydsl-jpa-codegen/src/test/java/com/querydsl/jpa/codegen/GenericExporterTest.java
package com.querydsl.jpa.codegen;

import java.io.File;

import org.junit.Test;

import com.querydsl.codegen.GenericExporter;

public class GenericExporterTest {

    @Test
    public void test() {
        GenericExporter exporter = new GenericExporter();
        exporter.setTargetFolder(new File("target/" + GenericExporterTest.class.getSimpleName()));
        exporter.export(getClass().getPackage());
    }
    
}
